client/ayon_hiero/api/plugin.py

```diff
@@ -882,13 +904,22 @@ def _convert_to_tag_data(self):
                         hero_data["productName"] = self.product_name + str(
                             self.track_index)
                     # in case track name and product name is the same then add
-                    if self.base_product_name == self.track_name:
+                    if self.subset_name == self.track_name:
                         hero_data["productName"] = self.product_name
                     # assign data to return hierarchy data to tag
                     tag_hierarchy_data = hero_data
 
         # add data to return data dict
-        return tag_hierarchy_data
+        self.tag_data.update(tag_hierarchy_data)
+
+        # add uuid to tag data
+        self.tag_data["uuid"] = str(uuid.uuid4())
+
+        # add review track only to hero track
+        if hero_track and self.review_layer:
+            self.tag_data.update({"reviewTrack": self.review_layer})
```

Comment:
Isn't this the same as this?
```suggestion
            self.tag_data["reviewTrack"] = self.review_layer
```